### PR TITLE
Implement childKeys method

### PR DESCRIPTION
**Pull request by alsamylkin**
Simply populate a slice, map and then combine the results.

### CODE REVIEW

File: value.go [hunk 1]
```diff
@@ -148,16 +148,20 @@ func (cv Value) WithDefault(value interface{}) Value {
...
 
-// TODO: Support enumerating child keys
-// 1. Add a method on Provider to get the child keys for a given prefix
-// 2. Implement in the various providers
-// 3. Merge the list here
-// 4. Return the set of keys.
-
-// ChildKeys returns the child keys
-// TODO(ai) what is this and do we need to keep it?
 func (cv Value) ChildKeys() []string {
```
Comment (anuptalwalkar) [inline value.go]: documentation?

File: value.go [hunk 2]
```diff
@@ -148,16 +148,20 @@ func (cv Value) WithDefault(value interface{}) Value {
...
-	return nil
+	var res []string
+
+	// intentionally ignoring the error, because we can't do much about it
+	// and simply return an empty slice in that case.
+	cv.Populate(&res)
+
+	var m map[string]interface{}
+	cv.Populate(&m)
+	for k := range m {
```
Comment (anuptalwalkar) [inline value.go]: `for k, _ := range m` for readability niceness

File: value.go [hunk 3]
```diff
@@ -148,16 +148,20 @@ func (cv Value) WithDefault(value interface{}) Value {
...
-
-// ChildKeys returns the child keys
-// TODO(ai) what is this and do we need to keep it?
 func (cv Value) ChildKeys() []string {
-	return nil
+	var res []string
+
+	// intentionally ignoring the error, because we can't do much about it
+	// and simply return an empty slice in that case.
+	cv.Populate(&res)
```
Comment (glibsm) [inline value.go]: i think you should check the error (to make errcheck happy) and if you incur it, just return an empty set or something? or nil

File: value.go (hunk 2 above)
Comment (glibsm) [inline value.go]: that syntax is a warning

File: static_provider_test.go [hunk 4]
```diff
@@ -271,3 +272,26 @@ func TestPopulateForMapsWithNotAssignableKeyTypes(t *testing.T) {
...
+	})
+
+	op := func(t *testing.T, key string, expected []string) {
+		keys := p.Get(key).ChildKeys()
+		sort.Strings(keys)
+		assert.Equal(t, expected, keys)
+	}
+
+	t.Run("Map", func(t *testing.T) { op(t, Root, []string{"one", "slice", "two"}) })
+	t.Run("Slice", func(t *testing.T) { op(t, "slice", []string{"13", "42"}) })
```
Comment (glibsm) [inline static_provider_test.go]: I don't get this. 

Why are we returning array values as child keys? If it was `[]string{0,1}` I would sort of understand, but even then I'd say lets just do maps, no?

Comment (alsamylkin) [inline static_provider_test.go]: Well, I was thinking about it as a collection of elements:
```yaml
- 13
- 42
```
but you right, the go way, would be {0, 1}

**Pull request merged**